src/Uno.UI/Controls/Window.macOS.cs

```diff
@@ -44,8 +44,210 @@ public Window(CGRect contentRect, NSWindowStyle aStyle, NSBackingStore buffering
+		///      give no examples.
+		/// 
+		/// The Export "method_name" attribute is fundamentally important to make this work, removing it will
+		/// break functionality and the method will never be called by macOS. Again, this seemly is because
+		/// Xamarin.macOS is not aware of it.
+		/// 
+		/// </remarks>
+		/// <param name="info">Information about the dragging session from the sender.</param>
+		/// <returns>The accepted drag operation(s).</returns>
+		[Export("draggingEntered:")] // Do not remove
+		public virtual NSDragOperation DraggingEntered(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				return DraggingEnteredAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				return NSDragOperation.None;
+			}
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingEntered(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DragEnter"/>.
+		/// </summary>
+		public Func<NSDraggingInfo, NSDragOperation> DraggingEnteredAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				return NSDragOperation.None;
+			};
+
+		/// <summary>
+		/// Method invoked when a drag operation updates over the <see cref="NSWindow"/>.
+		/// This is periodically called when the pointer is moved.
+		/// </summary>
+		/// <remarks>
+		/// See remarks in <see cref="DraggingEntered(NSDraggingInfo)"/>
+		/// </remarks>
+		/// <param name="draggingInfo">Information about the dragging session from the sender.</param>
+		[Export("draggingUpdated:")] // Do not remove
+		public virtual NSDragOperation DraggingUpdated(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				return DraggingUpdatedAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				return NSDragOperation.None;
+			}
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingUpdated(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DragOver"/>.
+		/// </summary>
+		public Func<NSDraggingInfo, NSDragOperation> DraggingUpdatedAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				return NSDragOperation.None;
+			};
+
+		/// <summary>
+		/// Method invoked when a drag operation ends.
+		/// </summary>
+		/// <remarks>
+		/// See remarks in <see cref="DraggingEntered(NSDraggingInfo)"/>
+		/// </remarks>
+		/// <param name="draggingInfo">Information about the dragging session from the sender.</param>
+		[Export("draggingEnded:")] // Do not remove
+		public virtual void DraggingEnded(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				DraggingEndedAction.Invoke(draggingInfo);
+			}
+			catch { }
+
+			return;
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingEnded(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DropCompleted"/>.
+		/// </summary>
+		public Action<NSDraggingInfo> DraggingEndedAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				return;
```

Comment:
Codacy found an issue: [Remove this redundant jump.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6364607)